### PR TITLE
audit-logs: move alerts partially to Thanos Rule

### PR DESCRIPTION
**Pull request by richardtief**
With Thanos, it is no longer necessary to have `kube-state metrics` available in multiple Prometheus. Kubernetes Prometheus keeps these metrics available so that they only need to be maintained in one place. Dependent services have been adjusted accordingly so that no absent metric alerts are generated.

### CODE REVIEW

File: system/audit-logs/vendor/fluent-audit-container/templates/prometheus-alerts.yaml [hunk 1]
```diff
@@ -12,11 +13,17 @@ metadata:
     app: fluent-audit-container-prometheus
     tier: infra
     type: alerting-rules
-    prometheus: {{ required ".Values.alerts.prometheus missing" $values.alerts.prometheus | quote }}
+    {{- if contains "thanos" $target }}
+    # alerts, which combine metrics from infra-collector and kubernetes, are available from the thanos metal query.
+    {{ $target }}: {{ index $values.alerts.prometheus $target | required (printf "$values.alerts.prometheus.%s missing" $target) }}
```
Comment (IvoGoman) [inline system/audit-logs/vendor/fluent-audit-container/templates/prometheus-alerts.yaml:18]: Can this be missing? The loop is only entered if there is any entry in `$values.alerts.prometheus`

File: system/audit-logs/values.yaml [hunk 2]
```diff
@@ -56,7 +56,10 @@ fluent_audit_container:
       host: none
   alerts:
     enabled: false
-    prometheus: infra-collector
+    # Keys = directory names in alerts/ and aggregations/
+    prometheus:
+      infra-collector: infra-collector
+      thanos-ruler: metal
```
Comment (IvoGoman) [inline system/audit-logs/values.yaml:62]: Maybe we can restructure this to not having to rely on parsing the name
```suggestion
    prometheus:
      - name: infra-collector
        type: prometheus
      - name: metal
        type: thanos-ruler
```

File: system/audit-logs/vendor/fluent-audit-container/templates/prometheus-alerts.yaml [hunk 3]
```diff
@@ -1,7 +1,8 @@
 {{- if eq .Values.global.clusterType "metal" }}
 {{- $values := .Values }}
 {{- if $values.alerts.enabled }}
-{{- range $path, $bytes := .Files.Glob "alerts/*.alerts" }}
+{{- range $target, $unused := $values.alerts.prometheus }}
+{{- range $path, $bytes := $.Files.Glob (printf "alerts/%s/*.alerts" $target) }}
```
Comment (IvoGoman) [inline system/audit-logs/vendor/fluent-audit-container/templates/prometheus-alerts.yaml:5]:
```suggestion
{{- range $path, $bytes := $.Files.Glob (printf "alerts/%s/*.alerts" $target.name) }}
```

File: system/audit-logs/vendor/fluent-audit-container/templates/prometheus-alerts.yaml [hunk 4]
```diff
@@ -12,11 +13,17 @@ metadata:
     app: fluent-audit-container-prometheus
     tier: infra
     type: alerting-rules
-    prometheus: {{ required ".Values.alerts.prometheus missing" $values.alerts.prometheus | quote }}
+    {{- if contains "thanos" $target }}
+    # alerts, which combine metrics from infra-collector and kubernetes, are available from the thanos metal query.
+    {{ $target }}: {{ index $values.alerts.prometheus $target | required (printf "$values.alerts.prometheus.%s missing" $target) }}
+    {{- else }}
+    prometheus: {{ index $values.alerts.prometheus $target | required (printf "$values.alerts.prometheus.%s missing" $target) }}
+    {{- end }}
```
Comment (IvoGoman) [inline system/audit-logs/vendor/fluent-audit-container/templates/prometheus-alerts.yaml:21]:
```suggestion
    {{ $target.type | required (printf "$values.alerts.prometheus.[%v].type missing" $i) }}: {{ $target.name | required (printf "$values.alerts.prometheus.[%v].name missing" $i) }}
```

File: system/audit-logs/vendor/fluent-audit-container/templates/prometheus-alerts.yaml [hunk 5]
```diff
@@ -1,7 +1,8 @@
 {{- if eq .Values.global.clusterType "metal" }}
 {{- $values := .Values }}
 {{- if $values.alerts.enabled }}
-{{- range $path, $bytes := .Files.Glob "alerts/*.alerts" }}
+{{- range $target, $unused := $values.alerts.prometheus }}
```
Comment (IvoGoman) [inline system/audit-logs/vendor/fluent-audit-container/templates/prometheus-alerts.yaml:4]:
```suggestion
{{- range $i, $target := $values.alerts.prometheus }}
```

**Review by IvoGoman — COMMENTED** [review]
I have added some comments w.r.t. using a list of dicts instead. I feel this makes it easier to understand, as all target are handled the same and there is no need for searching in strings. 
What do you think about it? (Have not rendered this locally yet)

File: system/audit-logs/vendor/fluent-audit-container/templates/prometheus-alerts.yaml (hunk 1 above)
Comment (richardtief) [reply system/audit-logs/vendor/fluent-audit-container/templates/prometheus-alerts.yaml:18]: This is just one solution that you can find on several charts in this repo. But yes - this is not failsafe, so we need to make sure it's there.

**Comment by richardtief** [comment]
Just rendered it locally and the thanos-ruler alert is not coming up. Only the prometheus: infra-collector one.

**Pull request merged**